gym_bridges/envs/bridges_env.py

```diff
@@ -261,11 +268,96 @@ def reset(self, state=None, gap_count=None):
+
+        x_window_coordinates = np.arange(
+            renderer_config.WINDOW_WIDTH // 2 - block_size * self.shape[1] // 2,
+            renderer_config.WINDOW_WIDTH // 2 + block_size * self.shape[1] // 2,
+            block_size,
+        )
+        y_window_coordinates = np.arange(
+            renderer_config.WINDOW_HEIGHT // 2 - block_size * self.shape[0] // 2,
+            renderer_config.WINDOW_HEIGHT // 2 + block_size * self.shape[0] // 2,
+            block_size,
+        )
+
+        for x_coordinate, x_window_coordinate in zip(
+            x_coordinates, x_window_coordinates
+        ):
+            for y_coordinate, y_window_coordinate in zip(
+                y_coordinates, y_window_coordinates
+            ):
+                rect = pygame.Rect(
+                    x_window_coordinate,
+                    y_window_coordinate,
+                    block_size,
+                    block_size,
+                )
+                if (
+                    self._state[y_coordinate][x_coordinate]
+                    == BridgesEnv.StateType.GROUND
+                ):
+                    screen.blit(textures[BridgesEnv.StateType.GROUND], rect)
+                elif (
+                    self._state[y_coordinate][x_coordinate]
+                    == BridgesEnv.StateType.BRICK
+                ):
+                    screen.blit(textures[BridgesEnv.StateType.BRICK], rect)
+                elif (
+                    self._state[y_coordinate][x_coordinate]
+                    == BridgesEnv.StateType.EMPTY
+                ):
+                    screen.blit(textures[BridgesEnv.StateType.EMPTY], rect)
+
     def render(self, mode="human"):
-        mapping = {
-            BridgesEnv.StateType.GROUND: "@@",
-            BridgesEnv.StateType.BRICK: "[]",
-            BridgesEnv.StateType.EMPTY: "  ",
-        }
-        flat_repr = tuple([mapping[x] for x in self._state.flatten()])
-        print((("%s" * self.shape[1] + "\n") * self.shape[0]) % flat_repr)
+        if mode == "human":
+            mapping = {
+                BridgesEnv.StateType.GROUND: "@@",
+                BridgesEnv.StateType.BRICK: "[]",
+                BridgesEnv.StateType.EMPTY: "  ",
+            }
+            flat_repr = tuple([mapping[x] for x in self._state.flatten()])
+            print((("%s" * self.shape[1] + "\n") * self.shape[0]) % flat_repr)
+        elif mode == "pygame":
+            if not self._pygame_initialized:
+                pygame.init()
+
+                self.screen = pygame.display.set_mode(
+                    (renderer_config.WINDOW_WIDTH, renderer_config.WINDOW_HEIGHT)
+                )
+                self.screen.fill(renderer_config.BLACK)
+
+                # Load the textures with a path relative to the game source code.
+                base_path = os.path.dirname(os.path.dirname(__file__))
+                textures_path = os.path.join(base_path, "renderer", "assets")
+
+                self.textures = {
```

Comment:
self._textures

Reply:
base on the chat, I think _pygame_initialized should stay in init and self._screen and self._textures can live in the scope of this function.

Reply:
(Full disclosure: My conventions weren't consistent when first writing this file so apologies for some inconsistency. Aiming for public attributes to not use _ and private ones to use _ )